prometheus/src/main/scala/filodb/prometheus/parse/Parser.scala

```diff
@@ -15,8 +19,52 @@ trait BaseParser extends Expressions with JavaTokenParsers with RegexParsers wit
     "[a-zA-Z_:][a-zA-Z0-9_:\\-\\.]*".r ^^ { str => Identifier(str) }
   }
 
-  protected lazy val labelValueIdentifier: PackratParser[Identifier] =
-    "([\"'])(?:\\\\\\1|.)*?\\1".r ^^ { str =>  Identifier(str.substring(1, str.size-1)) } //remove quotes
+  protected lazy val labelValueIdentifier: PackratParser[Identifier] = {
```

Comment:
Please add comment on why you had to write a custom parser here for posterity.

Reply:
Yes , please .

Reply:
Because it's easier to read than a regex with 6 backslashes in a row? ;-)